pkg/lib/cockpit-components-install-dialog.jsx

```diff
@@ -39,31 +39,42 @@ function format_to_fragments(fmt, arg) {
 }
 
 /* Calling install_dialog will open a dialog that lets the user
- * install the given package.
+ * install the given package on a OS that allows on-demand
+ * installation of packages. On an immutable OS, the dialog will
+ * instruct the user what to do.
  *
- * The install_dialog function returns a promise that is fulfilled when the dialog closes after
- * a successful installation.  The promise is rejected when the user cancels the dialog.
+ * The install_dialog function returns a promise that is fulfilled
+ * when the dialog closes after a successful installation.  The
+ * promise is rejected when the user cancels the dialog.
  *
- * If the package is already installed before the dialog opens, we still go
- * through all the motions and the dialog closes successfully without doing
- * anything when the use hits "Install".
- *
- * You shouldn't call install_dialog unless you know that PackageKit is available.
- * (If you do anyway, the resulting D-Bus errors will be shown to the user.)
+ * If the package is already installed before the dialog opens, we
+ * still go through all the motions and the dialog closes successfully
+ * without doing anything when the use hits "Install".
  */
 
 export function install_dialog(pkg, options) {
+    options = options || { };
+
+    if (!Array.isArray(pkg))
+        pkg = [pkg];
+
+    return PK.detect_with_details().then(details => {
+        if (details.available)
+            return install_dialog_mutable(pkg, options);
+        else if (details.reason == "immutable-os")
+            return install_dialog_immutable(pkg, options);
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18240-20230127-135755-f796be5b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)